src/main/java/org/somerville/swag/data/source/SQLiteExecute.java

```diff
@@ -12,29 +13,41 @@
 
     private DBConnection connection;
 
-    public SQLiteExecute(DBConnection connection) throws SQLException {
+    private LoggingService loggingService = LoggingServiceImpl.getInstance();
+
+    public SQLiteExecute(DBConnection connection) {
         this.connection = connection;
     }
 
     @Override
     public ResultSet executeSelect(String selectQuery) throws SQLStatementException {
-        ResultSet resultSet = null;
-        try (Connection connection = this.connection.connect()) {
-            Statement statement = connection.createStatement();
+        ResultSet resultSet;
+        try ( Statement statement = connection.connect().createStatement()) {
             resultSet = statement.executeQuery(selectQuery);
+            loggingService.logDatabaseSelectSuccess(selectQuery);
         } catch (SQLConnectionException | SQLException err) {
+            loggingService.logDatabaseSelectFailure(selectQuery, err.getMessage());
             throw new SQLStatementException(err.getMessage(), err);
         }
         return resultSet;
     }
 
     @Override
     public void executeInsert(String insertStatement) throws SQLStatementException {
-        try (Connection connection = this.connection.connect()) {
-            Statement statement = connection.createStatement();
+        try (Statement statement = connection.connect().createStatement()) {
```

Comment:
Really nice use of Try-With-Resources! Ensuring that objects are automatically closed when not needed!